kotest-extensions/src/jvmMain/kotlin/io/kotest/extensions/time/MutableClock.kt

```diff
@@ -0,0 +1,30 @@
+package io.kotest.extensions.time
+
+import java.io.Serializable
+import java.time.Clock
+import java.time.Instant
+import java.time.ZoneId
+
+class MutableClock(
+   private var instant: Instant,
+   private var zone: ZoneId,
+) : Clock(), Serializable {
+   fun withInstant(instant: Instant): Clock = MutableClock(instant, zone)
```

Comment:
This will return a new clock, rather than mutating this one.

Reply:
You're right, gonna change it